src/Categories/Morphism/Idempotent/Bundles.agda

```diff
@@ -0,0 +1,58 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category.Core
+
+-- Bundled versions of Idempotents, as well as maps between idempotents
+module Categories.Morphism.Idempotent.Bundles {o ℓ e} (𝒞 : Category o ℓ e) where
+
+open import Level
+
+open import Categories.Morphism.Idempotent 𝒞
+open import Categories.Morphism.Reasoning 𝒞
+
+open Category 𝒞
+open HomReasoning
+open Equiv
+
+--------------------------------------------------------------------------------
+-- Bundled Idempotents, and maps between them
+
+record BundledIdempotent : Set (o ⊔ ℓ ⊔ e) where
+  field
+    {obj} : Obj
+    isIdempotent : Idempotent obj
+
+  open Idempotent isIdempotent public
+
+open BundledIdempotent
+
+record Idempotent⇒ (I J : BundledIdempotent) : Set (ℓ ⊔ e) where
+  private
+    module I = BundledIdempotent I
+    module J = BundledIdempotent J
+  field
+    hom : I.obj ⇒ J.obj
+    absorbˡ : J.idem ∘ hom ≈ hom
+    absorbʳ : hom ∘ I.idem ≈ hom
+
+open Idempotent⇒
+
+--------------------------------------------------------------------------------
+-- Identity and Composition of maps between Idempotents
+
+idempotent⇒-id : ∀ {I} → Idempotent⇒ I I
```

Comment:
Here I'm much less decided, but: it's tempting to use just `id` (same for `∘` below), like is already done in many other places. Yes, this definitely causes clashes. But I feel like it might be better to leave the matter to the user, where they can choose between renaming, qualification or local modules as a way to disambiguate. Here you're forcing long names onto the user. Yes, they could rename to 'fix' this, but the reality is, few do it.